.circleci/config.yml

```diff
@@ -0,0 +1,38 @@
+version: 2
+jobs:
+  build:
+    machine: true
+    # Changing default working dir to the one used for the go path configuration
+    # Circle requires the use of the full path without env vars here
+    working_directory: /home/circleci/.go_workspace/src/github.com/hashicorp/terraform-aws-consul
+    steps:
+      - checkout
+      - run: echo 'export PATH=$HOME/.local/bin:$HOME/terraform:$HOME/packer:$PATH' >> $BASH_ENV
+      - restore_cache:
+          keys:
+            - dep-{{ checksum "test/Gopkg.lock" }}
+            - external-dep
+      # Install the gruntwork-module-circleci-helpers and use it to configure the build environment and run tests.
+      - run: curl -Ls https://raw.githubusercontent.com/gruntwork-io/gruntwork-installer/master/bootstrap-gruntwork-installer.sh | bash /dev/stdin --version v0.0.21
+      - run: gruntwork-install --module-name "gruntwork-module-circleci-helpers" --repo "https://github.com/gruntwork-io/module-ci" --tag "v0.12.2"
+      - run: gruntwork-install --module-name "build-helpers" --repo "https://github.com/gruntwork-io/module-ci" --tag "v0.12.2"
+      - run: gruntwork-install --module-name "aws-helpers" --repo "https://github.com/gruntwork-io/module-ci" --tag "v0.12.2"
+      - run: gruntwork-install --module-name "git-helpers" --repo "https://github.com/gruntwork-io/module-ci" --tag "v0.12.2"
+      - run: configure-environment-for-gruntwork-module --go-src-path test --use-go-dep --circle-ci-2
+      - save_cache:
+          key: external-dep
+          paths:
+            - $HOME/terraform
+            - $HOME/packer
+      - save_cache:
+          key: dep-{{ checksum "test/Gopkg.lock" }}
+          paths:
+            - $HOME/.go_workspace/src/github.com/hashicorp/terraform-aws-consul/test/vendor
+      - run:  if [[ $CIRCLE_BRANCH != "publish-amis" ]]; then run-go-tests --path test; fi
+      - deploy:
```

Comment:
Workflows would be ideal to control the flow regarding when or not to test or publish the amis. However because circle 2 runs each job in independent docker images, that would require either 

1. A complex sharing of the environments. Which is not possible because lots of the dependencies get installed at hardcoded directories like `usr/local/bin`
2. Gruntwork's premade docker image with the dependencies installed, which this repo does not have access because it is in a different github org.

So I chose to keep a good ol' `if`

Reply:
What does the GitHub org have to do with accessing a Docker image? I believe Docker auth is what we actually need here, and we have the proper AWS creds for ECR access. You just need to login...

Reply:
^ @mcalhoun